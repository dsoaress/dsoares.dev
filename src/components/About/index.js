import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'

import BlockContent from '../BlockContent'
import Social from '../Social'
import * as S from './styled'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <S.Wrapper>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <S.Grid>
        <S.TextWrapper>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </S.TextWrapper>
        <S.Image fluid={sanitySiteSettings.bio.image.asset.fluid} />
      </S.Grid>
    </S.Wrapper>
  )
}

export default About
