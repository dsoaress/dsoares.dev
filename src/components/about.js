import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

import BlockContent from './block-content'
import Social from './social'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <div>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <div>
        <div>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </div>
        <Img fluid={sanitySiteSettings.bio.image?.asset.fluid} />
      </div>
    </div>
  )
}

export default About
