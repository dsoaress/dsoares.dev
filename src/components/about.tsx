import { Block } from './block'

export function About() {
  return (
    <Block className="col-span-12 text-3xl leading-snug">
      <p className="text-neutral-400">
        Specializing in <span className="text-neutral-50">scalable, robust applications</span>, I
        focus on delivering seamless user experiences and optimizing backend systems for maximum
        efficiency.
      </p>
    </Block>
  )
}
