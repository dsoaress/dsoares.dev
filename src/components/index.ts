export * from './about'
export * from './block'
export * from './footer'
export * from './header'
export * from './home-container'
export * from './logo'
export * from './social'
