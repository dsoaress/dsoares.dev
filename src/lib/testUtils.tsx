import { render } from '@testing-library/react'
import { ComponentType, ReactElement, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles/theme'

type ProvidersProps = {
  children?: ReactNode
}

const Providers = ({ children }: ProvidersProps) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
