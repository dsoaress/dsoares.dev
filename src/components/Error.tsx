import { AiFillHome } from 'react-icons/ai'

import { Button } from '@/components/Button'
import { Container } from '@/components/Container'
import { Heading } from '@/components/Heading'
import { Link } from '@/components/Link'
import { Text } from '@/components/Text'
import { useData } from '@/hooks/useData'

type ErrorProps = {
  statusCode: number | string
  errorMessage: string
}

export function Error({ statusCode, errorMessage }: ErrorProps) {
  const { d } = useData()

  return (
    <main className="fixed inset-0 flex items-center justify-center">
      <Container>
        <div className="text-center">
          <Heading size="xl" className="mb-6">
            {statusCode}
          </Heading>

          <Text className="mb-4">{errorMessage}</Text>

          <Link href="/">
            <Button icon={AiFillHome}>{d.errors.buttonLabel}</Button>
          </Link>
        </div>
      </Container>
    </main>
  )
}
