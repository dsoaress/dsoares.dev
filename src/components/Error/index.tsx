import { AiFillHome } from 'react-icons/ai'

import { Button } from '@/components/Button'
import { Container } from '@/components/Container'
import { Heading } from '@/components/Heading'
import { Link } from '@/components/Link'
import { Text } from '@/components/Text'
import { useData } from '@/hooks/useData'

import { Content, Wrapper } from './styles'

type ErrorProps = {
  statusCode: number
  errorMessage: string
}

export function Error({ statusCode, errorMessage }: ErrorProps) {
  const { d } = useData()

  return (
    <Wrapper>
      <Container>
        <Content>
          <Heading size="xl" margin={6}>
            {statusCode}
          </Heading>

          <Text>{errorMessage}</Text>

          <Link href="/">
            <Button>
              <AiFillHome />
              {d.errors.buttonLabel}
            </Button>
          </Link>
        </Content>
      </Container>
    </Wrapper>
  )
}
