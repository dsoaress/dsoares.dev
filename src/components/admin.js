import { useState } from 'react'
import useSWR, { mutate } from 'swr'
import { signOut } from 'next-auth/client'

import Spinner from '@/components/spinner'
import { fetcher } from '@/lib/fetcher'

export default function Tasks() {
  const [label, setLabel] = useState('')
  const [url, setUrl] = useState('')

  const { data } = useSWR('/api', fetcher, {
    refreshInterval: 500
  })

  if (!data) return <Spinner />

  async function handleSubmit(e) {
    e.preventDefault()
    mutate('/api', { ...data, data: { links: { label, url } } }, false)

    await fetch('/api', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ link: { label, url } })
    })

    mutate('/api')
    setLabel('')
    setUrl('')
  }

  return (
    <div>
      <div>
        <button onClick={() => signOut()}>Sign out</button>
      </div>
      <form onSubmit={handleSubmit}>
        <div className="text-black">
          <label>Label</label>
          <input
            type="text"
            value={label}
            onChange={e => setLabel(e.target.value)}
            required
          />
          <label>Url</label>
          <input
            type="text"
            value={url}
            onChange={e => setUrl(e.target.value)}
            required
          />
        </div>
        <button type="submit">Add</button>
      </form>
      <div>
        {!data.links || data.links.length === 0 ? (
          <p>
            It looks like you don't have any more tasks... Would you like to add
            a new one?
          </p>
        ) : (
          data.links.map(link => (
            <div key={link._id}>
              <p>{link.label}</p>
              <p>{link.url}</p>
            </div>
          ))
        )}
      </div>
    </div>
  )
}
