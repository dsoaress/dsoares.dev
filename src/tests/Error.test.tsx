import mockRouter from 'next-router-mock'

import { Error } from '@/components/Error'
import { render, screen } from '@/lib/testUtils'

jest.mock('next/router', () => require('next-router-mock'))

describe('<Error />', () => {
  // beforeEach(() => {
  // })
  mockRouter.locale = 'en-us'

  it('Renders with status code and error message', () => {
    render(<Error statusCode={404} errorMessage="Error message" />)
    expect(screen.getByText('404')).toBeInTheDocument()
    expect(screen.getByText('404')).toMatchSnapshot()
    expect(screen.getByText('Error message')).toBeInTheDocument()
    expect(screen.getByText('Error message')).toMatchSnapshot()
  })
})
