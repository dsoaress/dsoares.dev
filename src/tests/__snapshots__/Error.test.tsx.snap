// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> Renders with status code and error message 1`] = `
<h1
  class="c-gMRxiR c-gMRxiR-ccsAlU-size-xl c-gMRxiR-cwookq-mb-6"
>
  404
</h1>
`;

exports[`<Error /> Renders with status code and error message 2`] = `
<p
  class="c-IbAAC"
>
  Error message
</p>
`;
